### PR TITLE
Setting up leaderboard

### PR DESCRIPTION
**Pull request by RiverRuby**
Previous branch was in some weird Git state, so creating a new PR here.

### CODE REVIEW

File: lib/zkProving.ts [hunk 1]
```diff
@@ -47,69 +109,20 @@ export const makeProofs = async (proverWasm: ProverWasm) => {
...
-  }
+  // load in previous sigmojis
+  const sigmojis = await loadSigmojis();
+  const sigmojisWithZKP = await Promise.all(
+    sigmojis.map((sigmoji) => addZKPToSigmoji(sigmoji, proverWasm, pubKeyTree))
+  );
+
+  // update localStorage
+  window.localStorage["sigmojis"] = JSON.stringify(
+    await Promise.all(sigmojisWithZKP.map(serializeSigmoji))
```
Comment (AndrewCLu) [inline lib/zkProving.ts]: Can chain this to the previous map

**Review by AndrewCLu — APPROVED** [review]
LGTM

File: components/modals/ProvingModal.tsx [hunk 2]
```diff
@@ -1,35 +1,126 @@
...
+
+    setProving(true);
+
+    const pubKeyTree = setupTree(wasm);
+    const sigmojis = await loadSigmojis();
+
+    // compute proofs in component to track progress for user
+    const wrappedSigmojis = sigmojis.map((sigmoji) =>
+      addZKPToSigmoji(sigmoji, wasm, pubKeyTree).then((result) => {
+        setCounter((prevCounter) => prevCounter + 1);
```
Comment (AndrewCLu) [inline components/modals/ProvingModal.tsx:82]: Maybe we only set the counter if this succeeds, i.e. sigmoji.ZKP !== ""

File: components/modals/ProvingModal.tsx [hunk 3]
```diff
@@ -1,35 +1,126 @@
...
+      setScore(sigmojis.length);
     }
 
     setup();
   });
 
+  const makeProofsWithCounter = async () => {
+    if (!wasm) return;
+
+    setProving(true);
```
Comment (AndrewCLu) [inline components/modals/ProvingModal.tsx]: setCounter(0);

File: lib/zkProving.ts (hunk 1 above)
Comment (RiverRuby) [reply lib/zkProving.ts]: refactored out this functionality

File: components/modals/ProvingModal.tsx (hunk 2 above)
Comment (RiverRuby) [reply components/modals/ProvingModal.tsx:82]: Ah I see. I guess we would also change the denominator of the proving percentage to be only the sigmojis that didn't already have ZKPs attached?

Comment (AndrewCLu) [reply components/modals/ProvingModal.tsx:82]: yeah or it could just start off as the percentage of sigmojis with existing proofs

**Pull request merged**